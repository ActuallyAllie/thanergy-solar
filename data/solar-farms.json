[
  {
    "id": 1,
    "name": "The Big Farm"
  },
  {
    "id": 2,
    "name": "Arcade Farm"
  },
  {
    "id": 3,
    "name": "The Talking Farms"
  },
  {
    "id": 4,
    "name": "Total Eclipse of the Solar Farm"
  },
  {
    "id": 5,
    "name": "Great Farms of Fire"
  },
  {
    "id": 6,
    "name": "The Farm and the Hound"
  },
  {
    "id": 7,
    "name": "Farm Inc."
  },
  {
    "id": 8,
    "name": "Bruce Farmstein & the E Street Farm"
  },
  {
    "id": 9,
    "name": "Farml√≥rien"
  },
  {
    "id": 10,
    "name": "Another Solar Farm"
  }
]