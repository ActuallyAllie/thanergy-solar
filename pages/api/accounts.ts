import accounts from '../../data/accounts.json';

export {
    accounts
};

export default function handler(req, res) {
    res.status(200).json(accounts);
}
