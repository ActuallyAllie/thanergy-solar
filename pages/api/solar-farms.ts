import solarFarms from '../../data/solar-farms.json';

export {
    solarFarms
};

export default function handler(req, res) {
    res.status(200).json(solarFarms);
}
