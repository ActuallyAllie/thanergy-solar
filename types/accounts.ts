export interface Account {
    id: number;
    customer_id: number;
    address: string;
    city: string;
    state: string;
    zip_code: string;
    solar_farm_id: number;
    capacity_share: number;
    created_date: string;
}
