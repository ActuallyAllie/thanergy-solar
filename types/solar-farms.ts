export interface SolarFarm {
    id: number;
    name: string;
}
