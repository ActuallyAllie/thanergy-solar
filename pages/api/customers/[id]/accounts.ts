import accounts from '../../../../data/accounts.json';

export function getAccountsByCustomerId(customerId: number) {
    return accounts.filter(a => a.customer_id === customerId);
}

export default function handler(req, res) {
    const accounts = getAccountsByCustomerId(Number(req.query.id));
    res.status(200).json(accounts);
}
