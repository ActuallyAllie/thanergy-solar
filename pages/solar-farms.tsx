import Head from 'next/head';
import { useRouter } from 'next/router';
import { accounts } from './api/accounts';
import { solarFarms } from './api/solar-farms';
import { Account, SolarFarm } from '../types/';
import { Container } from '../components/Container';
import { Table } from '../components/Table';
import { useMemo } from 'react';
import { useTable, useSortBy } from 'react-table';

interface Props {
    accounts: Account[];
    solarFarms: SolarFarm[];
}

export default function SolarFarms({ accounts, solarFarms }: Props) {
    const columns = useMemo(
        () => [
            {
                Header: 'ID',
                accessor: 'id',
            },
            {
                Header: 'Name',
                accessor: 'name',
            },
            {
                Header: 'Number of Accounts',
                accessor: (solarFarm: SolarFarm) =>
                    accounts.filter((a) => a.solar_farm_id === solarFarm.id)
                        .length,
                id: 'accounts',
            },
            {
                Header: 'Total Capacity Filled',
                accessor: (solarFarm: SolarFarm) =>
                    accounts
                        .filter((a) => a.solar_farm_id === solarFarm.id)
                        .reduce(
                            (prev, account) =>
                                !!account.capacity_share
                                    ? prev + account.capacity_share
                                    : prev,
                            0,
                        ),
                id: 'capacity',
            },
        ],
        [accounts],
    );

    const tableInstance = useTable({ columns, data: solarFarms }, useSortBy);

    return (
        <Container>
            <Head>
                <title>Solar Farms: Thanergy Solar</title>
                <meta
                    name="description"
                    content="Solar Farms harvesting the power of Dominicus!"
                />
            </Head>
            <Table title="Solar Farms" tableInstance={tableInstance} />
        </Container>
    );
}

export async function getServerSideProps() {
    return { props: { accounts, solarFarms } };
}
