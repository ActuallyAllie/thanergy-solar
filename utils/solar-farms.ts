import BigNumber from 'bignumber.js';
import { Account } from '../types/';

function getAccounts(accounts: Account[], solarFarmId: number) {
    return accounts.filter((a) => a.solar_farm_id === solarFarmId);
}

export function getNumberOfAccounts(accounts: Account[], solarFarmId: number) {
    return getAccounts(accounts, solarFarmId).length;
}

export function getTotalCapacityFilled(
    accounts: Account[],
    solarFarmId: number,
) {
    return getAccounts(accounts, solarFarmId).reduce(
        (prev, account) =>
            !!account.capacity_share ? prev.plus(account.capacity_share) : prev,
        new BigNumber(0),
    );
}
