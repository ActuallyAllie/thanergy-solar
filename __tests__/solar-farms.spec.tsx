/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SolarFarms from '../pages/solar-farms';
import { accounts } from '../pages/api/accounts';
import { solarFarms } from '../pages/api/solar-farms';

describe('Solar Farms Index', () => {
    it('renders a heading', () => {
        render(<SolarFarms accounts={accounts} solarFarms={solarFarms} />);

        const heading = screen.getByRole('heading', {
            name: /Solar Farms/i,
        });

        expect(heading).toBeInTheDocument();
    });

    it('renders solar farms unchanged', () => {
        const tree = renderer
            .create(<SolarFarms accounts={accounts} solarFarms={solarFarms} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
