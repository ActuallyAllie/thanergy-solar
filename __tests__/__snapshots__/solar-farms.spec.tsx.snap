// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solar Farms Index renders solar farms unchanged 1`] = `
<div
  className="xl:container mx-auto flex flex-col justify-center items-center"
>
  <div
    className="w-full h-auto flex-0 border-b border-solid border-gray-200 flex flex-col sm:flex-row items-center justify-between"
  >
    <div
      className="flex flex-row"
    >
      <div
        className="flex-0 px-5"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            alt="Thanergy solar logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="Thanergy solar logo"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fsolar-flare.png&w=128&q=75"
              srcSet="/_next/image?url=%2Fsolar-flare.png&w=64&q=75 1x, /_next/image?url=%2Fsolar-flare.png&w=128&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </div>
      </div>
      <div
        className="flex-1"
      >
        <h1
          className="text-2xl mt-3 text-4xl mt-2 text-indigo-500"
        >
          Thanergy Solar
        </h1>
      </div>
    </div>
    <div
      className="h-16 mb-3 sm:mb-0 sm:h-auto text-xl sm:text-base flex flex-col sm:flex-row justify-between text-center"
    >
      <a
        className="w-32 hover:text-blue-600 transition-all transform hover:scale-110 active:scale-100"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Customers
      </a>
      <a
        className="w-32 hover:text-blue-600 transition-all transform hover:scale-110 active:scale-100"
        href="/solar-farms"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Solar Farms
      </a>
    </div>
  </div>
  <main
    className="w-full py-5 flex flex-col flex-1 justify-center items-center"
  >
    <div
      className="w-full px-5"
    >
      <div
        className="w-full overflow-x-auto text-indigo-600 rounded-xl bg-gradient-to-r from-blue-50 to-indigo-100 p-3 md:p-10"
      >
        <h1
          className="text-3xl ml-4 mb-5 text-indigo-800"
        >
          Solar Farms
        </h1>
        <div
          className="w-full md:hidden"
        >
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                1
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                The Big Farm
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                9
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                5.1612
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100 bg-indigo-200"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                2
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Arcade Farm
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                6
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                3.2589
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                3
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                The Talking Farms
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                8
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                4.9628
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100 bg-indigo-200"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                4
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Total Eclipse of the Solar Farm
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                9
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                4.6806
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                5
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Great Farms of Fire
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                9
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                4.5865
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100 bg-indigo-200"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                6
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                The Farm and the Hound
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                7
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                4.736
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                7
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Farm Inc.
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                14
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                7.69
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100 bg-indigo-200"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                8
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Bruce Farmstein & the E Street Farm
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                9
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                5.0933
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                9
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Farmlórien
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                7
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                4.3104
              </li>
            </ul>
          </div>
          <div
            className="hover:bg-blue-200 hover:shadow-xl rounded p-5 my-2 active:shadow-none transition-all duration-100 bg-indigo-200"
            onClick={[Function]}
            role="row"
          >
            <ul>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  ID
                  :
                   
                </label>
                10
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Name
                  :
                   
                </label>
                Another Solar Farm
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Number of Accounts
                  :
                   
                </label>
                4
              </li>
              <li
                className=""
              >
                <label
                  className="font-bold mr-3"
                >
                  Total Capacity Filled
                  :
                   
                </label>
                3.326
              </li>
            </ul>
          </div>
        </div>
        <table
          className="w-full table-auto border-separate hidden md:table"
          role="table"
          style={
            Object {
              "borderSpacing": "0 0.5rem",
            }
          }
        >
          <thead>
            <tr
              role="row"
            >
              <th
                className="px-4 py-2 text-indigo-600"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                ID
                <span>
                  
                </span>
              </th>
              <th
                className="px-4 py-2 text-indigo-600"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Name
                <span>
                  
                </span>
              </th>
              <th
                className="px-4 py-2 text-indigo-600"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Number of Accounts
                <span>
                  
                </span>
              </th>
              <th
                className="px-4 py-2 text-indigo-600"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Total Capacity Filled
                <span>
                  
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                1
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                The Big Farm
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                9
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                5.1612
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100 bg-indigo-200"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                2
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Arcade Farm
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                6
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                3.2589
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                3
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                The Talking Farms
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                8
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4.9628
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100 bg-indigo-200"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Total Eclipse of the Solar Farm
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                9
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4.6806
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                5
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Great Farms of Fire
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                9
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4.5865
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100 bg-indigo-200"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                6
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                The Farm and the Hound
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                7
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4.736
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                7
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Farm Inc.
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                14
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                7.69
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100 bg-indigo-200"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                8
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Bruce Farmstein & the E Street Farm
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                9
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                5.0933
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                9
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Farmlórien
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                7
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4.3104
              </td>
            </tr>
            <tr
              className="hover:bg-blue-200 hover:shadow-xl active:shadow-none transition-all duration-100 bg-indigo-200"
              onClick={[Function]}
              role="row"
            >
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                10
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                Another Solar Farm
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                4
              </td>
              <td
                className="m-2 py-2 px-4"
                role="cell"
              >
                3.326
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </main>
  <footer
    className="w-full h-16 border-t border-solid border-gray-200 flex justify-center items-center"
  >
    <span
      className="text-pink-800"
    >
      Powered by Alecto
    </span>
  </footer>
</div>
`;
